polkadot/node/core/runtime-api/src/cache.rs

```diff
@@ -78,6 +78,7 @@ pub(crate) struct RequestResultCache {
 	backing_constraints: LruMap<(Hash, ParaId), Option<Constraints>>,
 	scheduling_lookahead: LruMap<SessionIndex, u32>,
 	validation_code_bomb_limits: LruMap<SessionIndex, u32>,
+	para_ids: LruMap<Hash, Vec<ParaId>>,
```

Comment:
Did you dig a bit to see if paras can only change once every session or more frequently than that?

If they can only change once per session we could cache them by the session index

Reply:
right, where should I look?

Reply:
Then let's cache by session index, like we do for scheduling_lookahead for example